pkg/problemdaemon/problem_daemon.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package problemdaemon
+
+import (
+	"github.com/golang/glog"
+
+	"k8s.io/node-problem-detector/pkg/types"
+)
+
+var (
+	handlers = make(map[types.ProblemDaemonType]types.ProblemDaemonHandler)
+)
+
+// Register registers a problem daemon factory method, which will be used to create the problem daemon.
+func Register(problemDaemonType types.ProblemDaemonType, handler types.ProblemDaemonHandler) {
+	handlers[problemDaemonType] = handler
+}
+
+// GetProblemDaemonNames retrieves the names of all available problem daemon types.
+func GetProblemDaemonNames() []types.ProblemDaemonType {
+	problemDaemons := []types.ProblemDaemonType{}
+	for problemDaemonType := range handlers {
+		problemDaemons = append(problemDaemons, problemDaemonType)
+	}
+	return problemDaemons
+}
+
+// GetProblemDaemonHandler retrieves the ProblemDaemonHandler for a specific type of problem daemon.
+func GetProblemDaemonHandler(problemDaemonType types.ProblemDaemonType) types.ProblemDaemonHandler {
+	return handlers[problemDaemonType]
```

Comment:
What happens when `problemDaemonType` is not in the map? This probably never happen for now. But later if we can disable some monitors, this needs to be checked. It is better to handle that now. Maybe panic?

Reply:
Good point. Done.